File: 2790-maximum-number-of-groups-with-increasing-length/2790-maximum-number-of-groups-with-increasing-length.cpp
class Solution {
public:
    int maxIncreasingGroups(vector<int>& A) {
        sort(A.begin(), A.end());
        long long tt = 0, ans = 0;
        for(int a : A) {
            tt += a;
            if(tt >= (ans+1)*(ans+2)/2) ++ans;
        }
        return ans;
    }
};

File: 2790-maximum-number-of-groups-with-increasing-length/2790-maximum-number-of-groups-with-increasing-length.java
class Solution {
    public int maxIncreasingGroups(List<Integer> A) {
        Collections.sort(A);
        long tt = 0, ans = 0;
        for(int a : A) {
            tt += a;
            if(tt >= (ans+1)*(ans+2)/2) ++ans;
        }
        return (int)ans;
    }
}